packages/wrangler/src/index.tsx

```diff
@@ -195,58 +195,94 @@ export async function main(argv: string[]): Promise<void> {
       const destination = path.join(process.cwd(), "wrangler.toml");
       if (fs.existsSync(destination)) {
         console.warn(`${destination} file already exists!`);
-        const result = await confirm(
+        const shouldContinue = await confirm(
           "Do you want to continue initializing this project?"
         );
-        if (!result) {
+        if (!shouldContinue) {
           return;
         }
+      } else {
+        const compatibilityDate = new Date().toISOString().substring(0, 10);
+        try {
+          await writeFile(
+            destination,
+            `compatibility_date = "${compatibilityDate}"` + "\n"
+          );
+          console.log(`✨ Successfully created wrangler.toml`);
+          // TODO: suggest next steps?
+        } catch (err) {
+          throw new Error(
+            `Failed to create wrangler.toml.\n${err.message ?? err}`
+          );
+        }
       }
 
-      const compatibilityDate = new Date().toISOString().substring(0, 10);
-      try {
-        await writeFile(
-          destination,
-          `compatibility_date = "${compatibilityDate}"` + "\n"
-        );
-        console.log(`✨ Successfully created wrangler.toml`);
-        // TODO: suggest next steps?
-      } catch (err) {
-        throw new Error(
-          `Failed to create wrangler.toml.\n${err.message ?? err}`
-        );
-      }
-
-      // if no package.json, ask, and if yes, create one
       let pathToPackageJson = await findUp("package.json");
-
       if (!pathToPackageJson) {
-        if (
-          await confirm("No package.json found. Would you like to create one?")
-        ) {
+        // If no package.json exists, ask to create one
+        const shouldCreatePackageJson = await confirm(
+          "No package.json found. Would you like to create one?"
+        );
+        if (shouldCreatePackageJson) {
           await writeFile(
             path.join(process.cwd(), "package.json"),
             JSON.stringify(
               {
                 name: "worker",
                 version: "0.0.1",
+                devDependencies: {
+                  wrangler: wranglerVersion,
+                },
               },
               null,
               "  "
             ) + "\n"
           );
+          await execa("npm", ["install", "--prefer-offline"], {
+            stdio: "inherit",
+          });
```

Comment:
My recommendation for making this testable (and reducing duplication) is to have a helper module (similar to what we did with cfetch) that exports the following function:

```ts
export async function npmInstall({deps = [], devDeps}: {deps?: string[], devDeps?: string[]} = {}) {
  if (devDeps) {
    await execa("npm", ["install", "--save-dev", ...devDeps], { stdio: "inherit" });
  }
  if (deps.length > 0 || !devDeps) {
    await execa("npm", ["install", ...deps], { stdio: "inherit" });
  }
}
```

Then we can mock out this module and replace it with a spy.

Reply:
Could do, but annoyingly that starts making the mock complicated as well, since it now has to modify package.json, and match other behaviour from `npm install`. I don't particularly like this myself at the moment (and tests could start failing when npm goes down) but it feels like the lesser of 2 evils right now.

Reply:
Yeah let's discuss this further for improvement in subsequent PRs.